### PR TITLE
ARROW-4528: [C++] Update lint docker container to LLVM-7

### PR DESCRIPTION
**Pull request by xhochy**

### CODE REVIEW

File: dev/lint/Dockerfile [hunk 1]
```diff
@@ -17,12 +17,16 @@
 
 FROM arrow:python-3.6
 
+RUN apt-get install -y -q gnupg
+RUN echo "deb https://apt.llvm.org/bionic/ llvm-toolchain-bionic-7 main" >> /etc/apt/sources.list
+RUN wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key| apt-key add -
+RUN apt update
```
Comment (kszucs) [inline dev/lint/Dockerfile]: @xhochy could You put it into a single RUN statement?

Comment (pitrou) [reply dev/lint/Dockerfile]: Why is this necessary? `clang-7` and friends are available in the official 18.04 repositories.

Comment (xhochy) [reply dev/lint/Dockerfile]: @kszucs Done.

@pitrou `clang-format-7` and the other tools are not in official Ubuntu repo.

Comment (pitrou) [reply dev/lint/Dockerfile]: Well, I've got `clang-format-7` here:
```
$ apt show clang-format-7
Package: clang-format-7
Version: 1:7-3~ubuntu0.18.04.1
Priority: optional
Section: universe/devel
Source: llvm-toolchain-7
Origin: Ubuntu
Maintainer: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>
Original-Maintainer: LLVM Packaging Team <pkg-llvm-team@lists.alioth.debian.org>
Bugs: https://bugs.launchpad.net/ubuntu/+filebug
Installed-Size: 2 029 kB
Depends: libc6 (>= 2.14), libllvm7 (= 1:7-3~ubuntu0.18.04.1), libstdc++6 (>= 5.2), python
Homepage: https://www.llvm.org/
Download-Size: 608 kB
APT-Manual-Installed: yes
APT-Sources: http://fr.archive.ubuntu.com/ubuntu bionic-updates/universe amd64 Packages
Description: Tool to format C/C++/Obj-C code
 Clang-format is both a library and a stand-alone tool with the goal of
 automatically reformatting C++ sources files according to configurable
 style guides. To do so, clang-format uses Clang's Lexer to transform an
 input file into a token stream and then changes all the whitespace around
 those tokens. The goal is for clang-format to both serve both as a user
 tool (ideally with powerful IDE integrations) and part of other
 refactoring tools, e.g. to do a reformatting of all the lines changed
 during a renaming.
 .
 This package also provides vim and emacs plugins.

```

Comment (xhochy) [reply dev/lint/Dockerfile]: My `apt` and `packages.ubuntu.com` skills failed me :(

**Comment by wesm** [comment]
+1. I removed the apt additions and tested locally

**Pull request closed without merging**